packages/api-format/src/codec/Metadata.ts

```diff
@@ -0,0 +1,291 @@
+// Copyright 2017-2018 @polkadot/api-format authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import CodecArray from './base/Array';
+import CodecEnum from './base/Enum';
+import CodecEnumType from './base/EnumType';
+import CodecOption from './base/Option';
+import CodecStruct from './base/Struct';
+import String from './String';
+import U16 from './U16';
+
+// Decodes the runtime metadata as passed through from the `state_getMetadata` call.
+export default class RuntimeMetadata extends CodecStruct<{
+  outerEvent: OuterEventMetadata,
+  modules: CodecArray<RuntimeModuleMetadata>
+}> {
+  constructor (value?: any) {
+    super({
+      outerEvent: OuterEventMetadata,
+      modules: CodecArray.with(RuntimeModuleMetadata)
```

Comment:
CodecArray.with are helper functions that just makes it simpler to define an array (and friends) in-place without having to do yet another class definition for a nested type.